ui/app/src/state/epics/pathNavigator.ts

```diff
@@ -126,6 +130,58 @@ export default [
       )
     ),
   // endregion
+  // region pathNavigatorBulkBackgroundRefresh
+  (action$, state$) =>
+    action$.pipe(
+      ofType(pathNavigatorBulkBackgroundRefresh.type, pathNavigatorBulkRefresh.type),
+      withLatestFrom(state$),
+      mergeMap(([{ payload }, state]) => {
+        const { ids } = payload;
+        let paths = [];
+        let optionsByPath = {};
+
+        ids.forEach((id) => {
+          const chunk = state.pathNavigator[id];
+          const { currentPath, keyword, limit, offset, excludes, sortStrategy, order } = chunk;
+          paths.push(currentPath);
+          optionsByPath[currentPath] = {
+            keyword,
+            limit,
+            offset,
+            excludes,
+            sortStrategy,
+            order
+          };
+        });
+
+        return forkJoin([
+          fetchItemsByPath(state.sites.active, paths, { castAsDetailedItem: true }),
+          fetchChildrenByPaths(state.sites.active, optionsByPath)
+        ]).pipe(
+          map(([items, children]) => {
+            const actions = [];
+            ids.forEach((id) => {
+              actions.push(
+                pathNavigatorFetchPathComplete({
+                  id,
+                  parent: items.find((item) => item.path.startsWith(withoutIndex(state.pathNavigator[id].currentPath))),
+                  children: children[state.pathNavigator[id].currentPath]
+                })
+              );
```

Comment:
Create an action that receives the bulk response and is processed accordingly in the reducer. Follow the action, actionComplete, actionFailed pattern.